src/pages/personal-settings/index.js

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useState, useContext } from 'react'
+import { Grid, Box } from '@mui/material'
+import { useFormik } from 'formik'
+import toast from 'react-hot-toast'
+import ErrorWindow from 'src/components/Shared/ErrorWindow'
+import WindowToolbar from 'src/components/Shared/WindowToolbar'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SelfServiceRepository } from 'src/repositories/SelfServiceRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { useResourceQuery } from 'src/hooks/resource'
+import { DataSets } from 'src/resources/DataSets'
+import * as yup from 'yup'
+
+const PersonalSettings = () => {
+  const [errorMessage, setErrorMessage] = useState(null)
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const [initialValues, setInitialValues] = useState({
+    recordId: null,
+    email: '',
+    activeStatus: '',
+    userType: '',
+    homePage: '',
+    username: '',
+    fullName: '',
+    languageId: '',
+    menuTheme: '',
+    languageName: ''
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+          const userData = window.sessionStorage.getItem('userData')
```

Comment:
Can we something regarding this? I know the team is copying this over from previous code but we need to refactor it into a utility. Maybe create a file having the session storage functions?